Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs

```diff
@@ -0,0 +1,174 @@
+using Content.Shared.Hands.Components;
+using Content.Shared.Interaction;
+using Robust.Shared.Containers;
+using Robust.Shared.Map;
+
+namespace Content.Shared.Hands.EntitySystems;
+
+public abstract partial class SharedHandsSystem : EntitySystem
+{
+    /// <summary>
+    ///     Checks if the contents of a hand is able to be removed from its container.
+    /// </summary>
+    public bool CanDrop(EntityUid uid, Hand hand, bool checkActionBlocker = true, SharedHandsComponent? hands = null)
+    {
+        if (hand.HeldEntity == null)
+            return false;
+
+        if (!hand.Container!.CanRemove(hand.HeldEntity.Value))
+            return false;
+
+        if (checkActionBlocker && !_actionBlocker.CanDrop(uid))
+            return false;
+
+        return true;
+    }
+
+    /// <summary>
+    ///     Attempts to drop the item in the currently active hand.
+    /// </summary>
+    public bool TryDrop(EntityUid uid, EntityCoordinates? targetDropLocation = null, bool checkActionBlocker = true, SharedHandsComponent? hands = null)
+    {
+        if (!Resolve(uid, ref hands))
+            return false;
+
+        if (hands.ActiveHand == null)
+            return false;
+
+        return TryDrop(uid, hands.ActiveHand, targetDropLocation, checkActionBlocker, hands);
+    }
+
+    /// <summary>
+    ///     Drops an item at the target location.
+    /// </summary>
+    public bool TryDrop(EntityUid uid, EntityUid entity, EntityCoordinates? targetDropLocation = null, bool checkActionBlocker = true, SharedHandsComponent? hands = null)
+    {
+        if (!Resolve(uid, ref hands))
+            return false;
+
+        if (!IsHolding(uid, entity, out var hand, hands))
+            return false;
+
+        return TryDrop(uid, hand, targetDropLocation, checkActionBlocker, hands);
+    }
+
+    /// <summary>
+    ///     Drops a hands contents at the target location.
+    /// </summary>
+    public bool TryDrop(EntityUid uid, Hand hand, EntityCoordinates? targetDropLocation = null, bool checkActionBlocker = true, SharedHandsComponent? hands = null)
+    {
+        if (!Resolve(uid, ref hands))
+            return false;
+
+        if (!CanDrop(uid, hand, checkActionBlocker, hands))
+            return false;
+
+        var entity = hand.HeldEntity!.Value;
+        DoDrop(uid, hand, hands);
+        _interactionSystem.DroppedInteraction(uid, entity);
```

Comment:
why cant `DoDrop` call dropped interaction? cause dropping into a container might not want to call that, or?

Reply:
can just be a bool argument in `DoDrop`

Reply:
Yeah containers and and moving stuff around the inventory shouldn't trigger some interactions.
But yeah it can just be a bool, ill do that.